jiant/trainer.py

```diff
@@ -328,23 +332,27 @@ def _setup_training(
                 batch_size=batch_size,
                 biggest_batch_first=True,
             )
-            tr_generator = iterator(task.train_data, num_epochs=None)
             task_info["iterator"] = iterator
+            tr_generator = iterator(task.train_data, num_epochs=None)
+            task_info["tr_generator"] = tr_generator
 
+            n_training_examples = task.n_train_examples
             if phase == "pretrain":
                 # Warning: This won't be precise when training_data_fraction is set, since each
                 #  example is included or excluded independently using a hashing function.
                 # Fortunately, it doesn't need to be.
-                task_info["n_tr_batches"] = math.ceil(
-                    task.n_train_examples * self._training_data_fraction / batch_size
-                )
-            else:
-                task_info["n_tr_batches"] = math.ceil(task.n_train_examples / batch_size)
+                n_training_examples *= self._training_data_fraction
+            task_info["n_tr_batches"] = math.ceil(n_training_examples / batch_size)
+            task_info["n_tr_steps"] = math.ceil(
+                task_info["n_tr_batches"] / self._accumulation_steps
+            )
 
-            task_info["tr_generator"] = tr_generator
             task_info["loss"] = 0.0
             task_info["total_batches_trained"] = 0
             task_info["n_batches_since_val"] = 0
```

Comment:
Make sure you've double-checked all existing uses of these two _batches_ variables. I'd expect that most of them should transition over to the new _steps_ variables.

Reply:
Confirmed — looks like they're accounted for.